src/api_server.cc

```diff
@@ -139,4 +147,28 @@ void MetadataApiServer::HandleMonitoredResource(
   }
 }
 
+void MetadataApiServer::HandleHealthz(
+    const HttpServer::request& request,
+    std::shared_ptr<HttpServer::connection> conn) {
+  if (health_checker_->IsHealthy()) {
+    if (config_.VerboseLogging()) {
+      LOG(INFO) << "Healthz returning 200";
+    }
+    conn->set_status(HttpServer::connection::ok);
+    conn->set_headers(std::map<std::string, std::string>({
+      {"Content-Type", "text/plain"},
+    }));
+    conn->write("healthy");
+  } else {
+    if (config_.VerboseLogging()) {
+      LOG(WARNING) << "Healthz returning 500";
+    }
+    conn->set_status(HttpServer::connection::internal_server_error);
+    conn->set_headers(std::map<std::string, std::string>({
+      {"Content-Type", "text/plain"},
+    }));
+    conn->write("unhealthy");
```

Comment:
Probably worth dumping the list of unhealthy components (and components whose callbacks returned `false`)...

Reply:
Done.